arduino-core/src/processing/app/helpers/FileUtils.java

```diff
@@ -209,25 +146,6 @@ public static String readFileToString(File file, String encoding) throws IOExcep
     }
   }
 
-  public static List<String> readFileToListOfStrings(File file) throws IOException {
-    List<String> strings = new LinkedList<>();
-    BufferedReader reader = null;
-    try {
-      reader = new BufferedReader(new FileReader(file));
-      String line;
-      while ((line = reader.readLine()) != null) {
-        line = line.replaceAll("\r", "").replaceAll("\n", "").replaceAll(" ", "");
```

Comment:
Uhm... does `Files.readAllLines` handles all types of newlines (MAC  / Linux / WIN)?
This `replaceAll` chain is quite weird since it remove also the spaces(!), I guess it has been added for some reason, let me check if the git history can give some hints.

Reply:
readAllLines(Path, Charset) states to handle them all:
![afbeelding](https://user-images.githubusercontent.com/9693840/55016778-bbea8e00-4fef-11e9-8625-02cfd69ec56e.png)
As for whitespaces, this method was only used to merge bootloader .hex (text) files with sketch .hex (text) files. This entire `MergeSketchWithBooloader.java` class does not seem to be used in anything other than a test. Is there a reason for keeping this code or could it be removed?

Reply:
The original commit is https://github.com/arduino/Arduino/commit/88e8019419, so yes it has been added specifically for MergeSketchWithBootloader. AFAIK it's not used anymore, so it can be removed.